// @react-navigation/elements/lib/commonjs/Button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Button = Button;
var _native = require("@react-navigation/native");
var _color = _interopRequireDefault(require("color"));
var React = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _PlatformPressable = require("./PlatformPressable");
var _Text = require("./Text");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
const BUTTON_RADIUS = 40;
function Button(props) {
  if ('screen' in props || 'action' in props) {
    return /*#__PURE__*/React.createElement(ButtonLink, props);
  } else {
    return /*#__PURE__*/React.createElement(ButtonBase, props);
  }
}
function ButtonLink(_ref) {
  let {
    screen,
    params,
    action,
    href,
    ...rest
  } = _ref;
  // @ts-expect-error: This is already type-checked by the prop types
  const props = (0, _native.useLinkProps)({
    screen,
    params,
    action,
    href
  });
  return /*#__PURE__*/React.createElement(ButtonBase, _extends({}, rest, props));
}
function ButtonBase(_ref2) {
  let {
    variant = 'tinted',
    color: customColor,
    android_ripple,
    style,
    children,
    ...rest
  } = _ref2;
  const {
    colors,
    fonts
  } = (0, _native.useTheme)();
  const color = customColor ?? colors.primary;
  let backgroundColor;
  let textColor;
  switch (variant) {
    case 'plain':
      backgroundColor = 'transparent';
      textColor = color;
      break;
    case 'tinted':
      backgroundColor = (0, _color.default)(color).fade(0.85).string();
      textColor = color;
      break;
    case 'filled':
      backgroundColor = color;
      textColor = (0, _color.default)(color).isDark() ? 'white' : (0, _color.default)(color).darken(0.71).string();
      break;
  }
  return /*#__PURE__*/React.createElement(_PlatformPressable.PlatformPressable, _extends({}, rest, {
    style: [{
      backgroundColor
    }, styles.button, style],
    android_ripple: {
      radius: BUTTON_RADIUS,
      color: (0, _color.default)(textColor).fade(0.85).string(),
      ...android_ripple
    }
  }), /*#__PURE__*/React.createElement(_Text.Text, {
    style: [{
      color: textColor
    }, fonts.regular, styles.text]
  }, children));
}
const styles = _reactNative.StyleSheet.create({
  button: {
    paddingHorizontal: 24,
    paddingVertical: 10,
    borderRadius: BUTTON_RADIUS
  },
  text: {
    fontSize: 14,
    lineHeight: 20,
    letterSpacing: 0.1,
    textAlign: 'center'
  }
});
