// @react-navigation/elements/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  Assets: true,
  Background: true,
  Button: true,
  getDefaultSidebarWidth: true,
  getDefaultHeaderHeight: true,
  getHeaderTitle: true,
  Header: true,
  HeaderBackButton: true,
  HeaderBackContext: true,
  HeaderBackground: true,
  HeaderButton: true,
  HeaderHeightContext: true,
  HeaderShownContext: true,
  HeaderTitle: true,
  useHeaderHeight: true,
  getLabel: true,
  Label: true,
  MissingIcon: true,
  PlatformPressable: true,
  ResourceSavingView: true,
  SafeAreaProviderCompat: true,
  Screen: true,
  Text: true
};
exports.Assets = void 0;
Object.defineProperty(exports, "Background", {
  enumerable: true,
  get: function () {
    return _Background.Background;
  }
});
Object.defineProperty(exports, "Button", {
  enumerable: true,
  get: function () {
    return _Button.Button;
  }
});
Object.defineProperty(exports, "Header", {
  enumerable: true,
  get: function () {
    return _Header.Header;
  }
});
Object.defineProperty(exports, "HeaderBackButton", {
  enumerable: true,
  get: function () {
    return _HeaderBackButton.HeaderBackButton;
  }
});
Object.defineProperty(exports, "HeaderBackContext", {
  enumerable: true,
  get: function () {
    return _HeaderBackContext.HeaderBackContext;
  }
});
Object.defineProperty(exports, "HeaderBackground", {
  enumerable: true,
  get: function () {
    return _HeaderBackground.HeaderBackground;
  }
});
Object.defineProperty(exports, "HeaderButton", {
  enumerable: true,
  get: function () {
    return _HeaderButton.HeaderButton;
  }
});
Object.defineProperty(exports, "HeaderHeightContext", {
  enumerable: true,
  get: function () {
    return _HeaderHeightContext.HeaderHeightContext;
  }
});
Object.defineProperty(exports, "HeaderShownContext", {
  enumerable: true,
  get: function () {
    return _HeaderShownContext.HeaderShownContext;
  }
});
Object.defineProperty(exports, "HeaderTitle", {
  enumerable: true,
  get: function () {
    return _HeaderTitle.HeaderTitle;
  }
});
Object.defineProperty(exports, "Label", {
  enumerable: true,
  get: function () {
    return _Label.Label;
  }
});
Object.defineProperty(exports, "MissingIcon", {
  enumerable: true,
  get: function () {
    return _MissingIcon.MissingIcon;
  }
});
Object.defineProperty(exports, "PlatformPressable", {
  enumerable: true,
  get: function () {
    return _PlatformPressable.PlatformPressable;
  }
});
Object.defineProperty(exports, "ResourceSavingView", {
  enumerable: true,
  get: function () {
    return _ResourceSavingView.ResourceSavingView;
  }
});
Object.defineProperty(exports, "SafeAreaProviderCompat", {
  enumerable: true,
  get: function () {
    return _SafeAreaProviderCompat.SafeAreaProviderCompat;
  }
});
Object.defineProperty(exports, "Screen", {
  enumerable: true,
  get: function () {
    return _Screen.Screen;
  }
});
Object.defineProperty(exports, "Text", {
  enumerable: true,
  get: function () {
    return _Text.Text;
  }
});
Object.defineProperty(exports, "getDefaultHeaderHeight", {
  enumerable: true,
  get: function () {
    return _getDefaultHeaderHeight.getDefaultHeaderHeight;
  }
});
Object.defineProperty(exports, "getDefaultSidebarWidth", {
  enumerable: true,
  get: function () {
    return _getDefaultSidebarWidth.getDefaultSidebarWidth;
  }
});
Object.defineProperty(exports, "getHeaderTitle", {
  enumerable: true,
  get: function () {
    return _getHeaderTitle.getHeaderTitle;
  }
});
Object.defineProperty(exports, "getLabel", {
  enumerable: true,
  get: function () {
    return _getLabel.getLabel;
  }
});
Object.defineProperty(exports, "useHeaderHeight", {
  enumerable: true,
  get: function () {
    return _useHeaderHeight.useHeaderHeight;
  }
});
var _Background = require("./Background");
var _Button = require("./Button");
var _getDefaultSidebarWidth = require("./getDefaultSidebarWidth");
var _getDefaultHeaderHeight = require("./Header/getDefaultHeaderHeight");
var _getHeaderTitle = require("./Header/getHeaderTitle");
var _Header = require("./Header/Header");
var _HeaderBackButton = require("./Header/HeaderBackButton");
var _HeaderBackContext = require("./Header/HeaderBackContext");
var _HeaderBackground = require("./Header/HeaderBackground");
var _HeaderButton = require("./Header/HeaderButton");
var _HeaderHeightContext = require("./Header/HeaderHeightContext");
var _HeaderShownContext = require("./Header/HeaderShownContext");
var _HeaderTitle = require("./Header/HeaderTitle");
var _useHeaderHeight = require("./Header/useHeaderHeight");
var _getLabel = require("./Label/getLabel");
var _Label = require("./Label/Label");
var _MissingIcon = require("./MissingIcon");
var _PlatformPressable = require("./PlatformPressable");
var _ResourceSavingView = require("./ResourceSavingView");
var _SafeAreaProviderCompat = require("./SafeAreaProviderCompat");
var _Screen = require("./Screen");
var _Text = require("./Text");
var _types = require("./types");
Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _types[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _types[key];
    }
  });
});
const Assets = exports.Assets = [
// eslint-disable-next-line import/no-commonjs
require('./assets/back-icon.png'),
// eslint-disable-next-line import/no-commonjs
require('./assets/back-icon-mask.png')];
