// @react-navigation/elements/lib/module/Header/HeaderButton.js
import * as React from 'react';
import { Platform, StyleSheet } from 'react-native';
import { PlatformPressable } from '../PlatformPressable';
export function HeaderButton(_ref) {
  let {
    disabled,
    onPress,
    pressColor,
    pressOpacity,
    accessibilityLabel,
    testID,
    style,
    href,
    children
  } = _ref;
  return /*#__PURE__*/React.createElement(PlatformPressable, {
    disabled: disabled,
    href: href,
    accessibilityLabel: accessibilityLabel,
    testID: testID,
    onPress: onPress,
    pressColor: pressColor,
    pressOpacity: pressOpacity,
    android_ripple: androidRipple,
    style: [styles.container, disabled && styles.disabled, style],
    hitSlop: Platform.select({
      ios: undefined,
      default: {
        top: 16,
        right: 16,
        bottom: 16,
        left: 16
      }
    })
  }, children);
}
const androidRipple = {
  borderless: true,
  foreground: Platform.OS === 'android' && Platform.Version >= 23,
  radius: 20
};
const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 8
  },
  disabled: {
    opacity: 0.5
  }
});
